diff -ruN heyu-2.0beta.6.2-original/setclock.c heyu-2.0beta.6.2/setclock.c
--- heyu-2.0beta.6.2-original/setclock.c	2006-05-27 11:40:58.000000000 +0200
+++ heyu-2.0beta.6.2/setclock.c	2006-09-04 10:03:08.000000000 +0200
@@ -375,7 +375,7 @@
 int c_stop_cm11a_poll ( int argc, char *argv[] )
 {	
    unsigned char data[3];
-   extern int    write(), xwrite(), check4poll();
+   extern int    xwrite(), check4poll();
    
    data[0] = 0x9b;
 
