diff -Nu -urNa dahdi-tools-2.2.0_rc3.ORIG/sethdlc.c dahdi-tools-2.2.0_rc3/sethdlc.c
--- dahdi-tools-2.2.0_rc3.ORIG/sethdlc.c	2008-08-27 18:13:25.000000000 +0200
+++ dahdi-tools-2.2.0_rc3/sethdlc.c	2009-06-03 18:21:16.000000000 +0200
@@ -35,6 +35,8 @@
 #include <dahdi/user.h>
 #include "dahdi_tools_version.h"
 
+#define ifreq __kernel_ifreq
+
 #if GENERIC_HDLC_VERSION != 4
 #error Generic HDLC layer version mismatch, please get correct sethdlc.c
 #endif
