Index: ruby-1.9.2-rc2/configure.in
===================================================================
--- ruby-1.9.2-rc2.orig/configure.in
+++ ruby-1.9.2-rc2/configure.in
@@ -1775,7 +1775,7 @@ if test "$with_dln_a_out" != yes; then
 	[linux* | gnu* | k*bsd*-gnu | netbsd* | bsdi* | kopensolaris*-gnu | freebsd7*], [
 			: ${LDSHARED='$(CC) -shared'}
 			if test "$rb_cv_binary_elf" = yes; then
-			    LDFLAGS="$LDFLAGS -Wl,-export-dynamic"
+			    LDFLAGS="$LDFLAGS -Wl,-export-dynamic -Wl,--no-undefined"
 			fi
 			rb_cv_dlopen=yes],
 	[interix*], [	: ${LDSHARED='$(CC) -shared'}
