Index: ruby-1.9.2-rc2/tool/rbinstall.rb
===================================================================
--- ruby-1.9.2-rc2.orig/tool/rbinstall.rb
+++ ruby-1.9.2-rc2/tool/rbinstall.rb
@@ -504,7 +504,6 @@ install?(:ext, :comm, :gem) do
 
   destdir = File.join(gpath, directories.grep(/^spec/)[0])
   default_gems = [
-    ['rake', 'rake.rb'],
     ['rdoc', 'rdoc.rb'],
     ['minitest', 'minitest/unit.rb'],
   ]
