diff -ur zaptel-1.4.1/Makefile zaptel-new/Makefile
--- zaptel-1.4.1/Makefile	2007-03-23 14:11:12.000000000 +1200
+++ zaptel-new/Makefile	2007-04-15 23:57:21.000000000 +1200
@@ -40,13 +40,20 @@
   BUILDVER:=linux26
 endif
 
+# This is not related to the version that we build. Rather, to the 
+# version that we runs. If we build for 2.4 using 2.4 headers on a 2.6
+# system with udev mounted on /dev , no point in installing files to /dev
+# because they'll be wiped at next reboot. XXX: should this be a use-flag? Perhaps: noudev?
+DYNFS:=$(shell ps ax | grep -v grep | grep -qw 'devfsd\|udevd' && echo "yes")
+
+# Check for udev rules directories
+ifneq (,$(wildcard $(ROOT_PREFIX)/etc/udev/rules.d))
+  UDEVRULES=yes
+endif
+
 # Set HOTPLUG_FIRMWARE=no to override automatic building with hotplug support
 # if it is enabled in the kernel.
 ifeq ($(BUILDVER),linux26)
-  ifneq (,$(wildcard $(INSTALL_PREFIX)/etc/udev/rules.d))
-    DYNFS=yes
-    UDEVRULES=yes
-  endif
   HOTPLUG_FIRMWARE:=$(shell if grep CONFIG_FW_LOADER $(KINCLUDES)/linux/autoconf.h | grep -q undef; then echo "no"; else echo "yes"; fi)
 else
   # Hotplug firmware loading is not supported on any other version then 2.6
@@ -136,7 +143,7 @@
   ALL_MODULES+=$(patsubst %,xpp/%.ko,xpp_usb xpd_fxo xpd_fxs xpp)
 endif
 
-CFLAGS+=-I. -O4 -g -Wall -DBUILDING_TONEZONE #-DTONEZONE_DRIVER
+CFLAGS+=-I. -Iinclude -g -Wall -DBUILDING_TONEZONE #-DTONEZONE_DRIVER
 ifneq (,$(findstring ppc,$(UNAME_M)))
 CFLAGS_PPC:=-fsigned-char
 endif
@@ -386,7 +393,7 @@
 $(UTILSO): %.o: %.c
 	$(CC) $(CFLAGS) -o $@ -c $<
 
-install: all devices install-modules install-libs install-include install-xpp
+install: devices install-modules install-libs install-include install-xpp
 ifeq ($(HOTPLUG_FIRMWARE),yes)
 	$(MAKE) -C firmware hotplug-install
 endif
@@ -401,20 +408,19 @@
 	if [ -f zttool ]; then \
 		$(INSTALL) -D -m 755 zttool $(INSTALL_PREFIX)/sbin/zttool; \
 	fi
+	install -d $(INSTALL_PREFIX)/usr/share/man/man8
 	$(INSTALL) -m 644 doc/ztcfg.8 $(INSTALL_PREFIX)/usr/share/man/man8
 	$(INSTALL) -m 644 doc/zttool.8 $(INSTALL_PREFIX)/usr/share/man/man8
-	[ `id -u` = 0 ] && /sbin/depmod -a $(KVERS) || :
+	[ -z "$(INSTALL_PREFIX)" -a `id -u` = 0 ] && /sbin/depmod -a $(KVERS) || :
 	[ -f $(CONFIG_FILE) ] || $(INSTALL) -D -m 644 zaptel.conf.sample $(CONFIG_FILE)
-	build_tools/genmodconf $(BUILDVER) "$(ROOT_PREFIX)" "$(filter-out zaptel ztdummy xpp zttranscode ztdynamic,$(BUILD_MODULES)) $(MODULE_ALIASES)"
-	@if [ -d /etc/modutils ]; then \
-		/sbin/update-modules ; \
-	fi
+	install -d -m 755 $(INSTALL_PREFIX)/etc/modules.d
+	build_tools/genmodconf $(BUILDVER) "$(INSTALL_PREFIX)" "$(filter-out zaptel ztdummy xpp zttranscode ztdynamic,$(BUILD_MODULES)) $(MODULE_ALIASES)"
 
 install-libs: $(LTZ_SO) $(LTZ_A)
 	$(INSTALL) -D -m 755 $(LTZ_A) $(LIB_DIR)/$(LTZ_A)
 	$(INSTALL) -D -m 755 $(LTZ_SO) $(LIB_DIR)/$(LTZ_SO).$(LTZ_SO_MAJOR_VER).$(LTZ_SO_MINOR_VER)
-	if [ -z "$(INSTALL_PREFIX)" -a `id -u` = 0 ]; then \
-		/sbin/ldconfig || : ;\
+	if [ `id -u` = 0 ]; then \
+		/sbin/ldconfig -n "$(INSTALL_PREFIX)"/usr/lib || : ;\
 	fi
 	rm -f $(LIB_DIR)$(LTZ_SO)
 	$(LN) -sf $(LTZ_SO).$(LTZ_SO_MAJOR_VER).$(LTZ_SO_MINOR_VER) \
@@ -425,7 +431,7 @@
 
 install-xpp:
 ifeq (yes,$(BUILD_XPP))
-	@$(MAKE) -C xpp/utils install
+	@$(MAKE) DESTDIR="$(INSTALL_PREFIX)" -C xpp/utils install
 endif
 
 install-include:
@@ -463,7 +469,8 @@
 else # DYNFS
   ifdef UDEVRULES
 	install -d $(INSTALL_PREFIX)/etc/udev/rules.d
-	build_tools/genudevrules > $(INSTALL_PREFIX)/etc/udev/rules.d/zaptel.rules
+	build_tools/genudevrules > zaptel.rules
+	install -D -m 644 zaptel.rules $(INSTALL_PREFIX)/etc/udev/rules.d/10-zaptel.rules
   else # !UDEVRULES
 	@echo "**** Dynamic filesystem detected -- not creating device nodes"
   endif
diff -ur zaptel-1.4.1/build_tools/genmodconf zaptel-new/build_tools/genmodconf
--- zaptel-1.4.1/build_tools/genmodconf	2007-02-01 03:12:37.000000000 +1300
+++ zaptel-new/build_tools/genmodconf	2007-04-15 23:10:20.000000000 +1200
@@ -79,20 +79,6 @@
 
 echo Building ${target}...
 
-if [ "${1}" = "linux24" ]; then
-    for mod in ${3}; do
-	if ! grep -q "post-install ${mod} " ${target}; then
-	    echo "post-install ${mod} /sbin/ztcfg" >> ${target}
-	fi
-    done
-elif [ "${1}" = "linux26" ]; then
-    for mod in ${3}; do
-	if ! grep -q "install ${mod} " ${target}; then
-	    echo "install ${mod} /sbin/modprobe --ignore-install ${mod} ${cmdopts} && /sbin/ztcfg" >> ${target}
-	fi
-    done
-fi
-
 if [ -z "${combined}" ]; then
     echo "***"
     echo "*** WARNING:"
diff -ur zaptel-1.4.1/build_tools/genudevrules zaptel-new/build_tools/genudevrules
--- zaptel-1.4.1/build_tools/genudevrules	2006-06-24 03:11:45.000000000 +1200
+++ zaptel-new/build_tools/genudevrules	2007-04-15 23:12:41.000000000 +1200
@@ -22,10 +22,10 @@
 
 cat <<EOF
 # zaptel devices with ownership/permissions for running as non-root
-KERNEL${match}"zapctl", NAME="zap/ctl", OWNER="asterisk", GROUP="asterisk", MODE="0660"
-KERNEL${match}"zaptranscode", NAME="zap/transcode", OWNER="asterisk", GROUP="asterisk", MODE="0660"
-KERNEL${match}"zaptimer", NAME="zap/timer", OWNER="asterisk", GROUP="asterisk", MODE="0660"
-KERNEL${match}"zapchannel", NAME="zap/channel", OWNER="asterisk", GROUP="asterisk", MODE="0660"
-KERNEL${match}"zappseudo", NAME="zap/pseudo", OWNER="asterisk", GROUP="asterisk", MODE="0660"
-KERNEL${match}"zap[0-9]*", NAME="zap/%n", OWNER="asterisk", GROUP="asterisk", MODE="0660"
+KERNEL${match}"zapctl", NAME="zap/ctl", OWNER="root", GROUP="dialout", MODE="0660"
+KERNEL${match}"zaptranscode", NAME="zap/transcode", OWNER="root", GROUP="dialout", MODE="0660"
+KERNEL${match}"zaptimer", NAME="zap/timer", OWNER="root", GROUP="dialout", MODE="0660"
+KERNEL${match}"zapchannel", NAME="zap/channel", OWNER="root", GROUP="dialout", MODE="0660"
+KERNEL${match}"zappseudo", NAME="zap/pseudo", OWNER="root", GROUP="dialout", MODE="0660"
+KERNEL${match}"zap[0-9]*", NAME="zap/%n", OWNER="root", GROUP="dialout", MODE="0660"
 EOF
