--- zaptel-1.4.8/Makefile.orig	2008-01-08 23:09:08.000000000 +0100
+++ zaptel-1.4.8/Makefile	2008-02-18 11:28:45.000000000 +0100
@@ -43,16 +43,24 @@ endif
 # Set HOTPLUG_FIRMWARE=no to override automatic building with hotplug support
 # if it is enabled in the kernel.
 ifeq ($(BUILDVER),linux26)
-  ifneq (,$(wildcard $(DESTDIR)/etc/udev/rules.d))
-    DYNFS=yes
-    UDEVRULES=yes
-  endif
   HOTPLUG_FIRMWARE:=$(shell if grep CONFIG_FW_LOADER $(KINCLUDES)/linux/autoconf.h | grep -q undef; then echo "no"; else echo "yes"; fi)
 else
   # Hotplug firmware loading is not supported on any other version then 2.6
   HOTPLUG_FIRMWARE:=no
 endif
 
+# This is not related to the version that we build. Rather, to the 
+# version that we runs. If we build for 2.4 using 2.4 headers on a 2.6
+# system with udev mounted on /dev , no point in installing files to /dev
+# because they'll be wiped at next reboot. XXX: should this be a use-flag? Perhaps: noudev?
+DYNFS:=$(shell ps ax | grep -v grep | grep -qw 'devfsd\|udevd' && echo "yes")
+
+# Check for udev rules directories
+ifneq (,$(wildcard $(ROOT_PREFIX)/etc/udev/rules.d))
+  UDEVRULES=yes
+endif
+
+
 ifeq ($(HOTPLUG_FIRMWARE),yes)
   CFLAGS+=-DHOTPLUG_FIRMWARE
 endif
@@ -145,7 +153,7 @@ else
   ALL_MODULES+=$(patsubst %,xpp/%.ko,xpp_usb xpd_fxo xpd_fxs xpp)
 endif
 
-OPTFLAG=-O2
+OPTFLAGS?=-O2
 CFLAGS+=-I. $(OPTFLAGS) -g -fPIC -Wall -DBUILDING_TONEZONE #-DTONEZONE_DRIVER
 ifneq (,$(findstring ppc,$(UNAME_M)))
 CFLAGS_PPC:=-fsigned-char
@@ -154,7 +162,7 @@ ifneq (,$(findstring x86_64,$(UNAME_M)))
 CFLAGS_x86_64:=-m64
 endif
 CFLAGS+=$(CFLAGS_PPC) $(CFLAGS_x86_64)
-KFLAGS=-I$(KINCLUDES) -O6
+KFLAGS=-I$(KINCLUDES) -O2
 KFLAGS+=-DMODULE -D__KERNEL__ -DEXPORT_SYMTAB -I$(KSRC)/drivers/net \
 	-Wall -I. -Wstrict-prototypes -fomit-frame-pointer -I$(KSRC)/drivers/net/wan -I$(KINCLUDES)/net
 ifneq (,$(wildcard $(KINCLUDES)/linux/modversions.h))
@@ -446,7 +454,7 @@ $(UTILS): %: %.o
 $(UTILSO): %.o: %.c
 	$(CC) $(CFLAGS) -o $@ -c $<
 
-install: all devices install-modules install-programs install-firmware
+install: devices install-modules install-programs install-firmware
 	@echo "###################################################"
 	@echo "###"
 	@echo "### Zaptel installed successfully."
@@ -471,10 +479,8 @@ endif
 
 # Pushing those two to a separate target that is not used by default:
 install-modconf:
-	build_tools/genmodconf $(BUILDVER) "$(ROOT_PREFIX)" "$(filter-out zaptel ztdummy xpp zttranscode ztdynamic,$(BUILD_MODULES)) $(MODULE_ALIASES)"
-	@if [ -d /etc/modutils ]; then \
-		/sbin/update-modules ; \
-	fi
+	install -d -m 755 $(DESTDIR)/etc/modules.d
+	build_tools/genmodconf $(BUILDVER) "$(DESTDIR)" "$(filter-out zaptel ztdummy xpp zttranscode ztdynamic,$(BUILD_MODULES)) $(MODULE_ALIASES)"
 
 install-firmware:
 ifeq ($(HOTPLUG_FIRMWARE),yes)
@@ -484,11 +490,9 @@ endif
 install-libs: libs
 	$(INSTALL) -D -m 755 $(LTZ_A) $(DESTDIR)$(LIB_DIR)/$(LTZ_A)
 	$(INSTALL) -D -m 755 $(LTZ_SO) $(DESTDIR)$(LIB_DIR)/$(LTZ_SO).$(LTZ_SO_MAJOR_VER).$(LTZ_SO_MINOR_VER)
-ifeq (,$(DESTDIR))
 	if [ `id -u` = 0 ]; then \
-		/sbin/ldconfig || : ;\
+		/sbin/ldconfig -n $(DESTDIR)$(LIB_DIR) || : ;\
 	fi
-endif
 	rm -f $(DESTDIR)$(LIB_DIR)$(LTZ_SO)
 	$(LN) -sf $(LTZ_SO).$(LTZ_SO_MAJOR_VER).$(LTZ_SO_MINOR_VER) \
 		$(DESTDIR)$(LIB_DIR)/$(LTZ_SO).$(LTZ_SO_MAJOR_VER)
@@ -535,7 +539,8 @@ ifndef DYNFS
 else # DYNFS
   ifdef UDEVRULES
 	install -d $(DESTDIR)/etc/udev/rules.d
-	build_tools/genudevrules > $(DESTDIR)/etc/udev/rules.d/zaptel.rules
+	build_tools/genudevrules > zaptel.rules
+	install -D -m 644 zaptel.rules $(DESTDIR)/etc/udev/rules.d/10-zaptel.rules
   else # !UDEVRULES
 	@echo "**** Dynamic filesystem detected -- not creating device nodes"
   endif
@@ -570,7 +575,7 @@ else
 	$(MAKE) -C datamods install
   endif
 endif
-	[ `id -u` = 0 ] && /sbin/depmod -a $(KVERS) || :
+	[ -z "$(DESTDIR)" -a `id -u` = 0 ] && /sbin/depmod -a $(KVERS) || :
 
 config:
 ifneq (,$(COPY_INITD))
--- zaptel-1.4.8/build_tools/genudevrules.orig	2007-12-16 18:43:20.000000000 +0100
+++ zaptel-1.4.8/build_tools/genudevrules	2008-02-18 11:15:48.000000000 +0100
@@ -31,5 +31,5 @@ KERNEL${match}"zappseudo", NAME="zap/pse
 KERNEL${match}"zap[0-9]*", NAME="zap/%n"
 
 # zaptel devices with ownership/permissions for running as non-root
-SUBSYSTEM=="zaptel",  OWNER="asterisk", GROUP="asterisk", MODE="0660"
+SUBSYSTEM=="zaptel",  OWNER="asterisk", GROUP="dialout", MODE="0660"
 EOF
diff -ur zaptel-1.4.4.orig/build_tools/genmodconf zaptel-1.4.4/build_tools/genmodconf
--- zaptel-1.4.4.orig/build_tools/genmodconf	2007-07-11 14:04:58.000000000 -0500
+++ zaptel-1.4.4/build_tools/genmodconf	2007-07-17 02:43:19.000000000 -0500
@@ -79,20 +79,6 @@
 
 echo Building ${target}...
 
-if [ "${1}" = "linux24" ]; then
-    for mod in ${3}; do
-	if ! grep -q "post-install ${mod} " ${target}; then
-	    echo "post-install ${mod} /sbin/ztcfg" >> ${target}
-	fi
-    done
-elif [ "${1}" = "linux26" ]; then
-    for mod in ${3}; do
-	if ! grep -q "install ${mod} " ${target}; then
-	    echo "install ${mod} /sbin/modprobe --ignore-install ${mod} ${cmdopts} && /sbin/ztcfg" >> ${target}
-	fi
-    done
-fi
-
 if [ -z "${combined}" ]; then
     echo "***"
     echo "*** WARNING:"
