diff -Naur yate.orig/Makefile.in yate/Makefile.in
--- yate.orig/Makefile.in	2010-04-13 14:03:59.000000000 +0200
+++ yate/Makefile.in	2010-04-13 17:12:57.000000000 +0200
@@ -7,12 +7,12 @@
 # override DEBUG at compile time to enable full debug or remove it all
 DEBUG :=
 
-CXX := @CXX@ -Wall
+CXX := @CXX@
 SED := sed
 DEFS :=
 LIBTHR:= @THREAD_LIB@
 INCLUDES := -I. -I@srcdir@
-CFLAGS := @CFLAGS@ @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@
 LDCONFIG:=true
 
@@ -31,7 +31,7 @@
 OBJS := main.o
 
 CLEANS = $(PROGS) $(SLIBS) $(LIBS) $(OBJS) yatepaths.h core
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CXX) $(LDFLAGS)
 
 DOCGEN_F := $(INCS)
diff -Naur yate.orig/clients/Makefile.in yate/clients/Makefile.in
--- yate.orig/clients/Makefile.in	2010-04-13 14:03:26.000000000 +0200
+++ yate/clients/Makefile.in	2010-04-13 14:06:39.000000000 +0200
@@ -7,7 +7,7 @@
 # override DEBUG at compile time to enable full debug or remove it all
 DEBUG :=
 
-CXX := @CXX@ -Wall
+CXX := @CXX@
 SED := sed
 DEFS :=
 MOC := @QT4_MOC@
@@ -17,7 +17,7 @@
 QT4_LIB_NET := @QT4_LIB_NET@
 LIBTHR:= @THREAD_LIB@
 INCLUDES := -I.. -I@top_srcdir@
-CFLAGS := @CFLAGS@ @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@
 YATELIBS := -L.. -lyate @LIBS@
 INCFILES := @top_srcdir@/yatengine.h @top_srcdir@/yatephone.h ../yateversn.h
@@ -44,7 +44,7 @@
 
 LOCALFLAGS =
 LOCALLIBS =
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CXX) $(LDFLAGS)
 
 ifneq (x@QT4_VER@,x)
diff -Naur yate.orig/clients/qt4/Makefile.in yate/clients/qt4/Makefile.in
--- yate.orig/clients/qt4/Makefile.in	2010-04-13 14:03:25.000000000 +0200
+++ yate/clients/qt4/Makefile.in	2010-04-13 16:59:07.000000000 +0200
@@ -4,7 +4,7 @@
 # override DEBUG at compile time to enable full debug or remove it all
 DEBUG :=
 
-CXX := @CXX@ -Wall
+CXX := @CXX@
 AR  := ar
 MOC := @QT4_MOC@
 QT4_INC := @QT4_INC@
@@ -14,7 +14,7 @@
 DEFS:=
 
 INCLUDES:=-I. -I@srcdir@ -I@top_srcdir@ $(QT4_INC)
-CXXFLAGS:=$(CXXFLAGS) @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS+= -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@
 YATELIBS := -L../.. -lyate @LIBS@
 INCFILES := @top_srcdir@/yateclass.h @top_srcdir@/yatecbase.h @srcdir@/qt4client.h
diff -Naur yate.orig/engine/Makefile.in yate/engine/Makefile.in
--- yate.orig/engine/Makefile.in	2010-04-13 14:03:42.000000000 +0200
+++ yate/engine/Makefile.in	2010-04-13 17:01:00.000000000 +0200
@@ -4,15 +4,15 @@
 # override DEBUG at compile time to enable full debug or remove it all
 DEBUG :=
 
-CC  := @CC@ -Wall
-CXX := @CXX@ -Wall
+CC  := @CC@
+CXX := @CXX@
 SED := sed
 DEFS :=
 LIBAUX:= @DLOPEN_LIB@ @LIBS@
 LIBTHR:= @THREAD_LIB@
 INCLUDES := -I.. -I@top_srcdir@
-CFLAGS := @CFLAGS@ @MODULE_CFLAGS@ @INLINE_FLAGS@
-CPPFLAGS := @CFLAGS@ @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CFLAGS += -Wall @MODULE_CFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@
 SONAME_OPT := @SONAME_OPT@
 LDCONFIG:=true
@@ -35,7 +35,7 @@
 LIBOBJS := $(CLSOBJS) $(ENGOBJS) $(TELOBJS) $(CLIOBJS)
 CLEANS = $(LIBOBJS) core
 CCOMPILE = $(CC) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CPPFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CXX) $(LDFLAGS)
 
 ifeq (@INTERNAL_REGEX@,yes)
diff -Naur yate.orig/libs/ilbc/Makefile.in yate/libs/ilbc/Makefile.in
--- yate.orig/libs/ilbc/Makefile.in	2010-04-13 14:03:11.000000000 +0200
+++ yate/libs/ilbc/Makefile.in	2010-04-13 17:02:17.000000000 +0200
@@ -1,12 +1,12 @@
 # Makefile
 # This file holds the make rules for the libilbc 
 
-CC  := @CC@ -Wall
+CC  := @CC@
 AR  := ar
 SED := sed
 DEFS :=
 INCLUDES := -I@top_srcdir@
-CFLAGS := @CFLAGS@ -O3 -funroll-loops -fomit-frame-pointer -fsigned-char @MODULE_CFLAGS@
+CFLAGS += -Wall @MODULE_CFLAGS@
 LDFLAGS:= @LDFLAGS@
 
 PROGS=
diff -Naur yate.orig/libs/yiax/Makefile.in yate/libs/yiax/Makefile.in
--- yate.orig/libs/yiax/Makefile.in	2010-04-13 14:03:11.000000000 +0200
+++ yate/libs/yiax/Makefile.in	2010-04-13 17:03:22.000000000 +0200
@@ -3,11 +3,11 @@
 
 DEBUG :=
 
-CXX  := @CXX@ -Wall
+CXX  := @CXX@
 AR  := ar
 DEFS :=
 INCLUDES := -I@top_srcdir@ -I../.. -I@srcdir@
-CFLAGS := @CFLAGS@ @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS := -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@
 YATELIBS := -L../.. -lyate @LIBS@
 INCFILES := @top_srcdir@/yateclass.h @srcdir@/yateiax.h
@@ -18,7 +18,7 @@
 
 LOCALFLAGS =
 LOCALLIBS =
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CXX) $(LDFLAGS)
 
 prefix = @prefix@
diff -Naur yate.orig/libs/yjabber/Makefile.in yate/libs/yjabber/Makefile.in
--- yate.orig/libs/yjabber/Makefile.in	2010-04-13 14:03:16.000000000 +0200
+++ yate/libs/yjabber/Makefile.in	2010-04-13 17:18:21.000000000 +0200
@@ -3,11 +3,11 @@
 
 DEBUG :=
 
-CXX  := @CXX@ -Wall
+CXX  := @CXX@
 AR  := ar
 DEFS :=
 INCLUDES := -I@top_srcdir@ -I../.. -I@srcdir@/../yxml -I@srcdir@
-CFLAGS := @CFLAGS@ @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@
 SONAME_OPT := @SONAME_OPT@
 YATELIBS := -L../.. -lyate @LIBS@
@@ -23,7 +23,7 @@
 
 LOCALFLAGS =
 LOCALLIBS =
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CXX) $(LDFLAGS)
 
 prefix = @prefix@
diff -Naur yate.orig/libs/ymgcp/Makefile.in yate/libs/ymgcp/Makefile.in
--- yate.orig/libs/ymgcp/Makefile.in	2010-04-13 14:03:09.000000000 +0200
+++ yate/libs/ymgcp/Makefile.in	2010-04-13 17:05:03.000000000 +0200
@@ -3,11 +3,11 @@
 
 DEBUG :=
 
-CXX  := @CXX@ -Wall
+CXX  := @CXX@
 AR  := ar
 DEFS :=
 INCLUDES := -I@top_srcdir@ -I../.. -I@srcdir@
-CFLAGS := @CFLAGS@ @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@
 SONAME_OPT := @SONAME_OPT@
 YATELIBS := -L../.. -lyate @LIBS@
@@ -22,7 +22,7 @@
 
 LOCALFLAGS =
 LOCALLIBS =
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CXX) $(LDFLAGS)
 
 prefix = @prefix@
diff -Naur yate.orig/libs/ymodem/Makefile.in yate/libs/ymodem/Makefile.in
--- yate.orig/libs/ymodem/Makefile.in	2010-04-13 14:03:12.000000000 +0200
+++ yate/libs/ymodem/Makefile.in	2010-04-13 17:07:15.000000000 +0200
@@ -3,11 +3,11 @@
 
 DEBUG :=
 
-CXX  := @CXX@ -Wall
+CXX  := @CXX@
 AR  := ar
 DEFS :=
 INCLUDES := -I@top_srcdir@ -I../.. -I@srcdir@
-CFLAGS := @CFLAGS@ @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@
 YATELIBS := -L../.. -lyate @LIBS@
 INCFILES := @top_srcdir@/yateclass.h @srcdir@/yatemodem.h
@@ -18,7 +18,7 @@
 
 LOCALFLAGS =
 LOCALLIBS =
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CXX) $(LDFLAGS)
 
 prefix = @prefix@
diff -Naur yate.orig/libs/ypbx/Makefile.in yate/libs/ypbx/Makefile.in
--- yate.orig/libs/ypbx/Makefile.in	2010-04-13 14:03:09.000000000 +0200
+++ yate/libs/ypbx/Makefile.in	2010-04-13 17:08:46.000000000 +0200
@@ -3,11 +3,11 @@
 
 DEBUG :=
 
-CXX  := @CXX@ -Wall
+CXX  := @CXX@
 AR  := ar
 DEFS :=
 INCLUDES := -I@top_srcdir@ -I../.. -I@srcdir@
-CFLAGS := @CFLAGS@ @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS := -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@
 YATELIBS := -L../.. -lyate @LIBS@
 INCFILES := @top_srcdir@/yateclass.h @srcdir@/yatepbx.h
@@ -18,7 +18,7 @@
 
 LOCALFLAGS =
 LOCALLIBS =
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CXX) $(LDFLAGS)
 
 prefix = @prefix@
diff -Naur yate.orig/libs/yrtp/Makefile.in yate/libs/yrtp/Makefile.in
--- yate.orig/libs/yrtp/Makefile.in	2010-04-13 14:03:17.000000000 +0200
+++ yate/libs/yrtp/Makefile.in	2010-04-13 17:09:30.000000000 +0200
@@ -3,11 +3,11 @@
 
 DEBUG :=
 
-CXX  := @CXX@ -Wall
+CXX  := @CXX@
 AR  := ar
 DEFS :=
 INCLUDES := -I@top_srcdir@ -I../.. -I@srcdir@
-CFLAGS := @CFLAGS@ @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS := -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@
 YATELIBS := -L../.. -lyate @LIBS@
 INCFILES := @top_srcdir@/yateclass.h @srcdir@/yatertp.h
@@ -18,7 +18,7 @@
 
 LOCALFLAGS =
 LOCALLIBS =
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CXX) $(LDFLAGS)
 
 prefix = @prefix@
diff -Naur yate.orig/libs/ysdp/Makefile.in yate/libs/ysdp/Makefile.in
--- yate.orig/libs/ysdp/Makefile.in	2010-04-13 14:03:09.000000000 +0200
+++ yate/libs/ysdp/Makefile.in	2010-04-13 17:20:17.000000000 +0200
@@ -3,11 +3,11 @@
 
 DEBUG :=
 
-CXX  := @CXX@ -Wall
+CXX  := @CXX@
 AR  := ar
 DEFS :=
 INCLUDES := -I@top_srcdir@ -I../.. -I@srcdir@
-CFLAGS := @CFLAGS@ @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@
 YATELIBS := -L../.. -lyate @LIBS@
 INCFILES := @top_srcdir@/yatemime.h @top_srcdir@/yatephone.h @srcdir@/yatesdp.h
@@ -18,7 +18,7 @@
 
 LOCALFLAGS =
 LOCALLIBS =
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CXX) $(LDFLAGS)
 
 prefix = @prefix@
diff -Naur yate.orig/libs/ysig/Makefile.in yate/libs/ysig/Makefile.in
--- yate.orig/libs/ysig/Makefile.in	2010-04-13 14:03:15.000000000 +0200
+++ yate/libs/ysig/Makefile.in	2010-04-13 17:10:43.000000000 +0200
@@ -3,12 +3,12 @@
 
 DEBUG :=
 
-CXX  := @CXX@ -Wall
+CXX  := @CXX@
 AR  := ar
 DEFS :=
 LIBTHR := @THREAD_LIB@
 INCLUDES := -I@top_srcdir@ -I../.. -I@srcdir@
-CFLAGS := @CFLAGS@ @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@
 SONAME_OPT := @SONAME_OPT@
 YATELIBS := -L../.. -lyate @LIBS@
@@ -28,7 +28,7 @@
 
 LOCALFLAGS =
 LOCALLIBS =
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CXX) $(LDFLAGS)
 
 prefix = @prefix@
diff -Naur yate.orig/libs/ysip/Makefile.in yate/libs/ysip/Makefile.in
--- yate.orig/libs/ysip/Makefile.in	2010-04-13 14:03:10.000000000 +0200
+++ yate/libs/ysip/Makefile.in	2010-04-13 17:11:21.000000000 +0200
@@ -3,11 +3,11 @@
 
 DEBUG :=
 
-CXX  := @CXX@ -Wall
+CXX  := @CXX@
 AR  := ar
 DEFS :=
 INCLUDES := -I@top_srcdir@ -I../.. -I@srcdir@
-CFLAGS := @CFLAGS@ @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@
 YATELIBS := -L../.. -lyate @LIBS@
 INCFILES := @top_srcdir@/yateclass.h @srcdir@/yatesip.h @srcdir@/util.h
@@ -18,7 +18,7 @@
 
 LOCALFLAGS =
 LOCALLIBS =
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CXX) $(LDFLAGS)
 
 prefix = @prefix@
diff -Naur yate.orig/libs/yxml/Makefile.in yate/libs/yxml/Makefile.in
--- yate.orig/libs/yxml/Makefile.in	2010-04-13 14:03:11.000000000 +0200
+++ yate/libs/yxml/Makefile.in	2010-04-13 17:12:06.000000000 +0200
@@ -3,11 +3,11 @@
 
 DEBUG :=
 
-CXX  := @CXX@ -Wall
+CXX  := @CXX@
 AR  := ar
 DEFS :=
 INCLUDES := -I@top_srcdir@ -I../.. -I@srcdir@
-CFLAGS := @CFLAGS@ @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@
 YATELIBS := -L../.. -lyate @LIBS@
 INCFILES := @top_srcdir@/yateclass.h @srcdir@/yatexml.h
@@ -18,7 +18,7 @@
 
 LOCALFLAGS =
 LOCALLIBS =
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CXX) $(LDFLAGS)
 
 prefix = @prefix@
diff -Naur yate.orig/modules/Makefile.in yate/modules/Makefile.in
--- yate.orig/modules/Makefile.in	2010-04-13 14:03:58.000000000 +0200
+++ yate/modules/Makefile.in	2010-04-13 17:14:27.000000000 +0200
@@ -7,8 +7,8 @@
 # override DEBUG at compile time to enable full debug or remove it all
 DEBUG :=
 
-CC  := @CC@ -Wall
-CXX := @CXX@ -Wall
+CC  := @CC@
+CXX := @CXX@
 MOC := @QT4_MOC@
 QT4_INC := @QT4_INC@
 QT4_LIB := @QT4_LIB@
@@ -17,8 +17,8 @@
 SED := sed
 DEFS :=
 INCLUDES := -I.. -I@top_srcdir@
-CFLAGS := @CFLAGS@ @MODULE_CFLAGS@ @INLINE_FLAGS@
-CPPFLAGS := @CFLAGS@ @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CFLAGS += -Wall @CFLAGS@ @MODULE_CFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -Wall @CFLAGS@ @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@
 YATELIBS:= -L.. -lyate @LIBS@
 MODFLAGS:= @MODULE_LDFLAGS@
@@ -140,7 +140,7 @@
 LOCALFLAGS =
 LOCALLIBS =
 CCOMPILE = $(CC) $(DEFS) $(DEBUG) $(INCLUDES) $(CFLAGS)
-COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CPPFLAGS)
+COMPILE = $(CXX) $(DEFS) $(DEBUG) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CXX) $(LDFLAGS)
 MODLINK = $(CXX) $(MODFLAGS) $(MODSTRIP) $(LDFLAGS)
 MODCOMP = $(COMPILE) $(MODFLAGS) $(MODSTRIP) $(LDFLAGS)
diff -Naur yate.orig/modules/test/Makefile.in yate/modules/test/Makefile.in
--- yate.orig/modules/test/Makefile.in	2010-04-13 14:03:57.000000000 +0200
+++ yate/modules/test/Makefile.in	2010-04-13 17:15:34.000000000 +0200
@@ -1,11 +1,11 @@
 # Makefile
 # This file holds the make rules for the Telephony Engine test cases
 
-CXX := @CXX@ -Wall
+CXX := @CXX@
 SED := sed
 DEFS :=
 INCLUDES := -I@top_srcdir@
-CFLAGS := -O0 @MODULE_CPPFLAGS@ @INLINE_FLAGS@
+CXXFLAGS += -O0 -Wall @MODULE_CPPFLAGS@ @INLINE_FLAGS@
 LDFLAGS:= @LDFLAGS@
 YATELIBS:= -L../.. -lyate @LIBS@
 MODFLAGS:= @MODULE_LDFLAGS@
@@ -18,7 +18,7 @@
 
 LOCALFLAGS =
 LOCALLIBS =
-COMPILE = $(CXX) $(DEFS) $(INCLUDES) $(CFLAGS)
+COMPILE = $(CXX) $(DEFS) $(INCLUDES) $(CXXFLAGS)
 LINK = $(CXX) $(LDFLAGS)
 MODLINK = $(CXX) $(MODFLAGS) $(MODSTRIP) $(LDFLAGS)
 MODCOMP = $(COMPILE) $(MODFLAGS) $(MODSTRIP) $(LDFLAGS)
