diff -Nu -burN yate/libs/ysdp/session.cpp yate.modified/libs/ysdp/session.cpp
--- yate/libs/ysdp/session.cpp
+++ yate.modified/libs/ysdp/session.cpp
@@ -293,7 +293,7 @@
 
     // no address means on hold or muted
     String origin;
-    origin << "yate " << m_sdpSession << " " << m_sdpVersion;
+    origin << "- " << m_sdpSession << " " << m_sdpVersion;
     origin << " IN IP4 " << (addr ? addr : m_host.safe());
     String conn;
     conn << "IN IP4 " << (addr ? addr : "0.0.0.0");
